core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java

```diff
@@ -91,48 +47,36 @@ public Object getFieldValue(Inode o) {
   @SuppressWarnings("unchecked")
   private IndexedSet<Inode> mChildren = new IndexedSet<Inode>(mIdIndex, mNameIndex);
 
-  private InodeDirectory(InodeDirectory.Builder builder) {
-    super(builder);
-    mMountPoint = builder.mMountPoint;
-  }
+  private boolean mMountPoint;
 
   /**
-   * Adds the given inode to the set of children.
+   * Creates a new instance of {@link InodeDirectory}.
    *
-   * @param child the inode to add
+   * @param id the id to use
    */
-  public synchronized void addChild(Inode child) {
-    mChildren.add(child);
+  public InodeDirectory(long id) {
+    super(id);
+    mDirectory = true;
+    mMountPoint = false;
+  }
+
+  private InodeDirectory(long id, long creationTimeMs) {
```

Comment:
it comes down to whether we want creation time to be immutable ... similarly for id, if we do not need it to be immutable, then we can just use the default constructor and add setters for id and creation time